flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/types/ResourceProfile.java

```diff
@@ -427,9 +432,7 @@ public ResourceProfile subtract(final ResourceProfile other) {
                             name,
                             (ignored, oldResource) -> {
                                 Resource resultResource = oldResource.subtract(resource);
-                                return resultResource.getValue().compareTo(BigDecimal.ZERO) == 0
-                                        ? null
-                                        : resultResource;
+                                return resultResource.isZero() ? null : resultResource;
```

Comment:
I think we no longer need to filter it in ahead after introducing the filter in constructor?